internal/db/resource.go

```diff
@@ -106,3 +111,61 @@ func (d *Database) ListResources(ctx context.Context, limit, offset int) (_ []Re
 	}
 	return resources, nil
 }
+
+// buildQuery is a utility function to build the database query according to two optional parameters.
+// namePrefixFilter: used to match resources name prefix.
+// typeFilter: used to match resources type. If this is not empty the resources are fetched from a single table.
+func buildQuery(db *gorm.DB, offset, limit int, namePrefixFilter, typeFilter string) (*gorm.DB, error) {
+	applicationOffersQuery := db.Select(selectApplicationOffers).
+		Model(&dbmodel.ApplicationOffer{}).
+		Where("(CASE WHEN ? = '' THEN TRUE ELSE application_offers.name LIKE ? END)", namePrefixFilter, namePrefixFilter+"%").
```

Comment:
```suggestion
		Where("(CASE WHEN ? = '' THEN TRUE ELSE application_offers.name LIKE ?% END)", namePrefixFilter, namePrefixFilter).
```
? and elsewhere

Reply:
it doesn't work with the SQL replacements